firstvoices/backend/views/dictionary_cleanup_views.py

```diff
@@ -0,0 +1,181 @@
+        site = self.get_validated_site()
+
+        qs = DictionaryCleanupJob.objects.filter(
+            site=site,
+            is_preview=False,
+            status__in=[JobStatus.COMPLETE, JobStatus.FAILED, JobStatus.CANCELLED],
+        )
+        qs.delete()
+
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+@extend_schema_view(
+    list=extend_schema(
+        description="A list of all dictionary cleanup preview results for the specified site.",
+        responses={
+            200: DictionaryCleanupPreviewJobSerializer,
+            403: OpenApiResponse(description=doc_strings.error_403),
+            404: OpenApiResponse(description=doc_strings.error_404_missing_site),
+        },
+        parameters=[site_slug_parameter],
+    ),
+    retrieve=extend_schema(
+        description="Details about a specific dictionary cleanup preview result.",
+        responses={
+            200: DictionaryCleanupPreviewJobSerializer,
+            403: OpenApiResponse(description=doc_strings.error_403),
+            404: OpenApiResponse(description=doc_strings.error_404_missing_site),
+        },
+        parameters=[
+            site_slug_parameter,
+            id_parameter,
+        ],
+    ),
+    create=extend_schema(
+        description="Create and queue a new dictionary cleanup preview job.",
+        responses={
+            202: DictionaryCleanupPreviewJobSerializer,
+            403: OpenApiResponse(description=doc_strings.error_403),
+            404: OpenApiResponse(description=doc_strings.error_404_missing_site),
+        },
+        parameters=[site_slug_parameter],
+    ),
+    clear=extend_schema(
+        description="Deletes all finished dictionary cleanup preview results for the specified site.",
+        responses={
+            204: OpenApiResponse(description=doc_strings.success_204_deleted),
+            403: OpenApiResponse(description=doc_strings.error_403),
+            404: OpenApiResponse(description=doc_strings.error_404_missing_site),
+        },
+        parameters=[site_slug_parameter],
+    ),
+)
+class DictionaryCleanupPreviewViewSet(
+    DictionaryCleanupJobViewSet,
+):
+    serializer_class = DictionaryCleanupPreviewJobSerializer
+
+    def get_queryset(self):
+        site = self.get_validated_site()
+        return (
+            DictionaryCleanupJob.objects.filter(site=site, is_preview=True)
+            .select_related("site", "created_by", "last_modified_by")
+            .order_by("created")
+        )
+
+    def perform_create(self, serializer):
+        # Create the model instance with the preview flag set
+        instance = serializer.save(is_preview=True)
+
+        # Queue the cleanup task after model creation
+        transaction.on_commit(
+            lambda: cleanup_dictionary.apply_async(
+                (instance.id,), link_error=link_error_handler.s()
+            )
+        )
+
+    @action(methods=["delete"], detail=False)
+    def clear(self, request, *args, **kwargs):
```

Comment:
While we are making changes here, since this API changes multiple resources, in REST terms it should be a POST not a DELETE